test/features/resource_availability.feature

```diff
@@ -16,184 +16,166 @@ Feature: Re-identification risk governance acknowledgement or Resource visibilit
         Then I press "save"
         And I wait for 10 seconds
         Then I execute the script "document.getElementById('field-image-url').value='http://ckanext-data-qld.docker.amazee.io/'"
-        Then I fill in "name" with "res1"
-        Then I fill in "description" with "description"
-        Then I fill in "size" with "1024"
-        Then I select "Resource NOT visible/Pending acknowledgement" from "resource_visibility"
+        Then I fill in "name" with "Resource hidden"
+        Then I fill in "description" with "hide resource"
+        Then I select "TRUE" from "resource_visibility"
+        Then I select "NO" from "governance_acknowledgement"
         Then I press the element with xpath "//button[@value='go-metadata']"
         And I wait for 10 seconds
         Then I should see "Data and Resources"
```

Comment:
Tests can be run in any order, they don't need to be run sequentially, the way you written this is that if an earlier test fails it will fail a later test, or if later test is run on its own it will fail due to missing data.

You can collapse those visibility tests to the end, note you can't do examples loop as you can't create what already exists, so just add the logout, persona swap, login to the following scenarios:
Scenario: Sysadmin creates dataset with Contains de-identified data is YES and Resource visibility is TRUE and Acknowledgement is NO
Scenario: Sysadmin creates dataset with Contains de-identified data is NO and Resource visibility is TRUE and Acknowledgement is NO
Scenario: Sysadmin creates dataset with Contains de-identified data is NO and Resource visibility is FALSE and Acknowledgement is NO
Scenario: Sysadmin creates dataset with Contains de-identified data is YES and Resource visibility is TRUE and Acknowledgement is YES

example code below, this could be wrapped in two(2) python 'step'  functions, one for visible to all and visible to editor+ see -> https://github.com/qld-gov-au/ckan-qld-infrastructure/blob/master/test/features/steps/steps.py#L133 


```
       Then I log out <!-- this step may need to be created
       And I go to "/contains-de-identified-data-yes-visibility-true-acknowledgment-no" <!-- change to created dataset
       Then I should not see "Resource hidden"  <!-- this is anon test update to see or not see depending on scenario under test  (also include viewing of resource fields if visible)
       Then I log out <!-- this step may need to be created
       And "User" as the persona 
       And I log in
       And I go to "/contains-de-identified-data-yes-visibility-true-acknowledgment-no" <!-- change to created dataset
       Then I should not see "Resource hidden"  <!-- this is logged in 'user' test update to see or not see depending on scenario under test (also include viewing of resource fields if visible)
       And "TestOrgAdmin" as the persona 
       And I log in
       And I go to "/contains-de-identified-data-yes-visibility-true-acknowledgment-no" <!-- change to created dataset
       And I wait for 3 seconds
       Then I press the element with xpath "//a[@title='Resource hidden']"
       Then I should not see "Visibility/Governance Acknowledgment"
       Then I should see "Resource visible"
       Then I should see "Re-identification risk governance completed?" 
       Then I log out <!-- this step may need to be create
       And "TestOrgEditor" as the persona 
       And I log in
       And I go to "/contains-de-identified-data-yes-visibility-true-acknowledgment-no" <!-- change to created dataset
       And I wait for 3 seconds
       Then I press the element with xpath "//a[@title='Resource hidden']"
       Then I should not see "Visibility/Governance Acknowledgment"
       Then I should see "Resource visible"
       Then I should see "Re-identification risk governance completed?" 
       Then I log out <!-- this step may need to be create
       And "SysAdmin" as the persona 
       And I log in
       And I go to "/contains-de-identified-data-yes-visibility-true-acknowledgment-no" <!-- change to created dataset
       And I wait for 3 seconds
       Then I press the element with xpath "//a[@title='Resource hidden']"
       Then I should not see "Visibility/Governance Acknowledgment"
       Then I should see "Resource visible"
       Then I should see "Re-identification risk governance completed?" 
       Then I log out <!-- this step may need to be create

```